libpod/options.go

```diff
@@ -2397,3 +2400,32 @@ func WithPodCPUSetCPUs(inp string) PodCreateOption {
 		return nil
 	}
 }
+
+func WithPodPidNS(inp specgen.Namespace) PodCreateOption {
+	return func(p *Pod) error {
+		if p.valid {
+			return define.ErrPodFinalized
+		}
+		if p.config.UsePodPID {
+			ns := specs.LinuxNamespace{}
+			ns.Type = "pid"
```

Comment:
Is there a reason we need to recreate the namespace?

Reply:
I think `specs.LinuxNameSpace{}` is the type that the infra generator takes versus `specgen.NameSpace`, we need to switch over at some point and this is behind the scenes enough.

Reply:
Ahhh, I missed the type change. Hm. My concern would be forwards-compat - we need to make sure that pods from before this change, without a valid namespace in the DB, continue to work.

Reply:
well,  the `specgen.ParseNamespace()` function I use in pkg/domain/entities/pods.go takes the type specgen.NameSpace so the actual validation and parsing relies more on the specgen version this is just a conversion. This switch will only execute when someone explicitly says `--pid=...` so would forwards-compat be a problem?